perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -260,51 +266,124 @@ def GenerateJobFileString(disk, against_device,
       iodepths=io_depths))
 
 
-def GetOrGenerateJobFileString(fio_jobfile, disk, against_device,
-                               scenario_strings, io_depths, working_set_size):
+FILENAME_PARAM_REGEXP = re.compile('filename\s*=.*$', re.MULTILINE)
+
+
+def ProcessedJobFileString(fio_jobfile, remove_filename):
+  """Get the contents of a job file as a string, slightly edited.
+
+  Args:
+    fio_jobfile: the path to a fio job file.
+    remove_filename: bool. If true, remove the filename parameter from
+      the job file.
+
+  Returns:
+    The job file as a string, possibly without filename parameters.
+  """
+
+  with open(fio_jobfile, 'r') as jobfile:
+    if remove_filename:
+      return FILENAME_PARAM_REGEXP.sub('', jobfile.read())
+    else:
+      return jobfile.read()
+
+
+def GetOrGenerateJobFileString(job_file_path, scenario_strings,
+                               against_device, disk,
+                               io_depths, working_set_size):
   """Get the contents of the fio job file we're working with.
 
   This will either read the user's job file, if given, or generate a
   new one.
 
   Args:
-    fio_jobfile: string or None. The path to the user's jobfile, if provided.
-    disk: the disk.BaseDisk object we're benchmarking with.
-    against_device: bool. True if we're using a raw disk.
-    scenario_strings: list of strings or None. The workload scenarios to
-      generate.
-    io_depths: iterable. The IO queue depths to test.
-    working_set_size: int or None. If int, the size of the working set in GB.
+    job_file_path: string or None. The path to the user's job file, if
+      provided.
+    scenario_strings: list of strings or None. The workload scenarios
+      to generate.
+    against_device: bool. True if testing against a raw device, False
```

Comment:
I passed them as parameters because I think it makes the functions easier to understand and because I think it is much easier to mock parameters than flags, which makes these functions easier to test. Do you think it's worth it to use flags rather than parameters?